Ethernet library compile warning

Reposted from main IDE.  IDE 1.8.10.  Controllino MEGA, simple ethernet test sketch......

Get this on compile....

In file included from C:\Program Files (x86)\Arduino\libraries\Ethernet\src\Dns.cpp:8:0:

C:\Program Files (x86)\Arduino\libraries\Ethernet\src\Dns.cpp: In member function 'uint16_t DNSClient::BuildRequest(const char*)':

C:\Program Files (x86)\Arduino\libraries\Ethernet\src\utility/w5100.h:457:25: warning: result of '(256 << 8)' requires 18 bits to represent, but 'int' only has 16 bits [-Wshift-overflow=]

 #define htons(x) ( (((x)<<8)&0xFF00) | (((x)>>8)&0xFF) )

                      ~~~^~~

C:\Program Files (x86)\Arduino\libraries\Ethernet\src\Dns.cpp:164:18: note: in expansion of macro 'htons'

  twoByteBuffer = htons(QUERY_FLAG | OPCODE_STANDARD_QUERY | RECURSION_DESIRED_FLAG);

                  ^~~~~

--- a/src/utility/w5100.h
+++ b/src/utility/w5100.h
@@ -455,14 +455,23 @@ extern W5100Class W5100;
 #define UTIL_H
 
 #ifndef htons
+// The host order of the Arduino platform is little endian.
+// Sometimes it is desired to convert to big endian (or
+// network order)
 
-#define htons(x) ( (((x)<<8)&0xFF00) | (((x)>>8)&0xFF) )
+// Host to Network short
+#define htons(x) ( (((x)&0xFF)<<8) | (((x)>>8)&0xFF) )
+
+// Network to Host short
 #define ntohs(x) htons(x)
 
+// Host to Network long
 #define htonl(x) ( ((x)<<24 & 0xFF000000UL) | \
                    ((x)<< 8 & 0x00FF0000UL) | \
                    ((x)>> 8 & 0x0000FF00UL) | \
                    ((x)>>24 & 0x000000FFUL) )
+
+// Network to Host long
 #define ntohl(x) htonl(x)
 
 #endif // !defined(htons)

